contracts/libraries/TickBitmap.sol

```diff
@@ -13,7 +13,7 @@ library TickBitmap {
     /// @return bitPos The bit position in the word where the flag is stored
     function position(int24 tick) private pure returns (int16 wordPos, uint8 bitPos) {
         wordPos = int16(tick >> 8);
-        bitPos = uint8(tick % 256);
+        bitPos = uint8(uint24(tick % 256));
```

Comment:
I used `uint24` as the intermediate type because this is what the implicit conversion does on 0.7.6 but this actually looks like a bug to me. When `wordPos` is negative the result will not be correct. Maybe it's not supposed to be negative but then it should be `uint16`, not `int16`.

For example, if `tick` is `-10 * 256 + 2`, the function will decode it into `-10` and `254`, not `-10` and `2` as I would expect it to based on the description.